cli/cmd/generate.go

```diff
@@ -384,6 +396,32 @@ func addWorkloadOwnerKeyToManifest(manifst *manifest.Manifest, keyPath string) e
 	return nil
 }
 
+func addSeedshareOwnerKeyToManifest(manifst *manifest.Manifest, keyPath string) error {
+	keyData, err := os.ReadFile(keyPath)
+	if err != nil {
+		return fmt.Errorf("reading seedshare owner key: %w", err)
+	}
+	block, _ := pem.Decode(keyData)
+	if block == nil {
+		return errors.New("failed to decode PEM block")
+	}
+	var publicKey []byte
+	switch block.Type {
+	case "PUBLIC KEY":
+		publicKey = block.Bytes
+	case "RSA PRIVATE KEY":
+		privateKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+		if err != nil {
+			return fmt.Errorf("parsing RSA private key: %w", err)
+		}
+		publicKey = x509.MarshalPKCS1PublicKey(&privateKey.PublicKey)
+	default:
+		return fmt.Errorf("unsupported PEM block type: %s", block.Type)
+	}
+	manifst.SeedshareOwnerPubKeys = append(manifst.SeedshareOwnerPubKeys, manifest.NewHexString(publicKey))
```

Comment:
It would be good if we could use `manifest.MarshalSeedShareOwnerKey` so that we have a single place where algorithms are chosen.

Reply:
We should follow up and also move the workload owner key stuff into the manifest package.